### PR TITLE
 feat(#28): makes wip prefix configurable.

### PR DESCRIPTION
**Pull request by hemanik**
This PR brings in the following features to the WIP plugin.

- Adds some common, work in progress prefixes (WIP, DO NOT MERGE, DON'T MERGE, WORK-IN-PROGRESS case insensitive with combinations of `( )`, `[ ]`, `:`) as default to be added to the PR title. 
- Allow for user defined prefixes by making it configurable by setting the `title_prefixes` property in the configuration file for the work in progress plugin.  

Fixes: #28

### CODE REVIEW

**Comment by MatousJobanek** [comment]
- [ ] Please add also the description to the documentation

File: pkg/plugin/work-in-progress/event_handler_gh_test.go [hunk 1]
```diff
@@ -80,8 +84,31 @@ var _ = Describe("Test Keeper Plugin features", func() {
 			Ω(err).ShouldNot(HaveOccurred())
 		})
 
+		It("should mark opened PR as work-in-progress when prefixed with configured prefix", func() {
```
Comment (MatousJobanek) [inline pkg/plugin/work-in-progress/event_handler_gh_test.go]: `should mark opened PR as work-in-progress when title starts with configured prefix`

File: pkg/plugin/work-in-progress/configuration_gh_test.go [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+				Owner:    "owner",
+				RepoName: "repo",
+				Hash:     "46cb8fac44709e4ccaae97448c65e8f7320cfea7",
+			}
+
+			// when
+			configuration := wip.LoadConfiguration(logger, change)
+
+			// then
+			Expect(configuration.Prefix).To(ConsistOf("[work in progress]", "work in progress"))
```
Comment (MatousJobanek) [inline pkg/plugin/work-in-progress/configuration_gh_test.go]: please verify also that the `combine_defaults` is equal true

File: pkg/plugin/work-in-progress/event_handler.go [hunk 3]
```diff
@@ -40,6 +38,7 @@ type GitHubWIPPRHandler struct {
 
 var (
 	handledPrActions = []string{"opened", "reopened", "edited"}
+	defaultPrefixes  = []string{"WIP", "DO NOT MERGE", "DON'T MERGE", "WORK-IN-PROGRESS"}
```
Comment (dipak-pawar) [inline pkg/plugin/work-in-progress/event_handler.go]: @hemanik Are we missing `WORK IN PROGRESS` here?

Comment (MatousJobanek) [reply pkg/plugin/work-in-progress/event_handler.go]: according to the requested changes in the [issue](https://github.com/arquillian/ike-prow-plugins/issues/28#issuecomment-382414676) no. According to common sense, we could miss it, but I would say wait for some time to get the idea which patterns (prefixes) are used in the projects. After all, it is still configurable

Comment (dipak-pawar) [reply pkg/plugin/work-in-progress/event_handler.go]: @MatousJobanek I totally understand your point of this is configurable. But I think `Work In Progress` is common pattern which you can see in general

Comment (MatousJobanek) [reply pkg/plugin/work-in-progress/event_handler.go]: Personally, I use only `WIP` or `do not merge`. But that is my opinion. 
If you are used to using this prefix as well, then feel free to open a PR with this change ;-)

**Pull request merged**